features/docs/stap-definities.feature

```diff
@@ -0,0 +1,326 @@
+      | persoon      | INSERT INTO public.lo3_pl_persoon(stapel_nr,volg_nr,persoon_type,burger_service_nr,geslachts_naam,onjuist_ind) VALUES($1,$2,$3,$4,$5,$6) | 0,1,P,000000012,Jansen,O |
+      |              | INSERT INTO public.lo3_pl_persoon(stapel_nr,volg_nr,persoon_type,geslachts_naam) VALUES($1,$2,$3,$4)                                     | 0,0,P,Janssen            |
+
+  Rule: Gegeven de persoon met burgerservicenummer '<bsn>' heeft de volgende '<naam-pl-tabel>' gegevens
+
+    Scenario: Persoon heeft 'inschrijving' gegevens
+      Gegeven de persoon met burgerservicenummer '000000012' heeft de volgende 'inschrijving' gegevens
+      | aanduiding uitgesloten kiesrecht (38.10) |
+      | A                                        |
+      Dan zijn de gegenereerde SQL statements
+      | key          | text                                                                                                                                                 | values          |
+      | inschrijving | INSERT INTO public.lo3_pl(pl_id,mutatie_dt,kiesrecht_uitgesl_aand) VALUES((SELECT MAX(pl_id)+1 FROM public.lo3_pl),current_timestamp,$1) RETURNING * | A               |
+      | persoon      | INSERT INTO public.lo3_pl_persoon(stapel_nr,volg_nr,persoon_type,burger_service_nr) VALUES($1,$2,$3,$4)                                              | 0,0,P,000000012 |
+
+    Scenario: Persoon heeft 'verblijfstitel' gegevens
+      Gegeven de persoon met burgerservicenummer '000000012' heeft de volgende 'verblijfstitel' gegevens
+      | aanduiding verblijfstitel (39.10) |
+      | 37                                |
+      Dan zijn de gegenereerde SQL statements
+      | key            | text                                                                                                                                     | values          |
+      | inschrijving   | INSERT INTO public.lo3_pl(pl_id,mutatie_dt,geheim_ind) VALUES((SELECT MAX(pl_id)+1 FROM public.lo3_pl),current_timestamp,$1) RETURNING * | 0               |
+      | persoon        | INSERT INTO public.lo3_pl_persoon(stapel_nr,volg_nr,persoon_type,burger_service_nr) VALUES($1,$2,$3,$4)                                  | 0,0,P,000000012 |
+      | verblijfstitel | INSERT INTO public.lo3_pl_verblijfstitel(volg_nr,verblijfstitel_aand) VALUES($1,$2)                                                      | 0,37            |
+
+  Rule: En de/het '<naam-pl-tabel>' is gewijzigd naar de volgende gegevens
+
+    Scenario: Persoon heeft gewijzigde 'verblijfstitel' gegevens
+      Gegeven de persoon met burgerservicenummer '000000012' heeft de volgende 'verblijfstitel' gegevens
+      | aanduiding verblijfstitel (39.10) |
+      | 37                                |
+      En de 'verblijfstitel' is gewijzigd naar de volgende gegevens
```

Comment:
voor verblijfstitel is dit functioneel meestal geen juiste omschrijving. In de meeste gevallen komt er niet een wijziging op de verblijfstitel, maar een nieuwe verblijfstitel. Denk aan mensen met een tijdelijke verblijfsvergunning. Wanneer de einddatum van de verblijfsvergunning (bijna) bereikt wordt, krijgt de persoon een nieuwe verblijfstitel met een nieuwe begin- en einddatum.
Het is dan verwarrend (vanuit business/domein perspectief) om te spreken over het wijzigen van de verblijfstitel.

Reply:
ik zou hier dus een ander voorbeeld kiezen, zoals verblijfplaats of gezag.

Voor verblijfstitel kan je een synonieme stap maken die exact hetzelfde doet, maar met andere tekst.

Eigenlijk werkt in de werkelijkheid verblijfstitel net zo als partner: je kan er achter elkaar meerdere hebben, echter maar één tegelijk. 
Alleen in het datamodel is het mogelijk om meerdere partners tegelijkertijd te hebben en worden verschillende partners daarom op een stapel geplaatst, terwijl het datamodel voor een verblijfstitel maar één actuele tegelijk mogelijk maakt. 
Voor verblijfstitels kunnen volgnummers worden gebruikt worden alsof het stapels zijn (nieuwe verblijfstitel) of als volgnummers (bij wijziging of correctie). 

Dus functioneel is het juister om de Gegevenstappen van meervoudige categorieën (zoals partners) te gebruiken, alleen met daarachter de implementatie van enkelvoudige categorieën.

Ik denk dat je dat niet wilt, dus moet het net iets anders zijn. Zoiets?:
```
Gegeven de persoon met burgerservicenummer '000000012' heeft een 'verblijfstitel' verkregen met de volgende gegevens
En de persoon heeft een 'verblijfstitel' verkregen met de volgende gegevens
```

Voor wijzigingen en correcties op verblijfstitels kan je wat mij betreft wel gewoon de stappen van enkelvoudige categorieën gebruiken:
```
En de 'verblijfstitel' is gewijzigd naar de volgende gegevens
En de 'verblijfstitel' is gecorrigeerd naar de volgende gegevens
```

Reply:
In de verblijfstitel feature zie ik een scenario dat een verblijfstitel geen datum einde hoeft te hebben. Hoe moet je dan aangeven als op een later moment datum einde wordt gezet? Wordt de verblijfstitel dan niet gewijzigd?

Reply:
Ik heb de twee verblijfstitel stappen toegevoegd. De implementatie nog niet.

Reply:
Dit is nu geïmplementeerd